### PR TITLE
Add block support to AzureBlobStoreRepositoryTests

### PR DESCRIPTION
**Pull request by tlrx**
This commit adds support for Put Block API to the internal HTT server
used in Azure repository integration tests. This allows to test the 
behavior of the Azure SDK client when the Azure Storage service 
returns errors when uploading Blob in multiple blocks or when 
downloading a blob using ranged downloads.

### CODE REVIEW

**Review by original-brownbear — APPROVED** [review]
LGTM :) Thanks again for this one!

**Comment by tlrx** [comment]
Jenkins test this

**Comment by tlrx** [comment]
@elasticmachine update branch

**Comment by tlrx** [comment]
Thanks @original-brownbear !

File: plugins/repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureBlobStoreRepositoryTests.java [hunk 1]
```diff
@@ -90,7 +109,43 @@ protected Settings nodeSettings(int nodeOrdinal) {
     }
 
     /**
-     * AzureRepositoryPlugin that allows to set very low values for the Azure's client retry policy
+     * Test the snapshot and restore of an index which has large segments files.
+     */
+    public void testSnapshotWithLargeSegmentFiles() throws Exception {
```
[the remark this answers is not in the document]
Comment (tlrx) [reply plugins/repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureBlobStoreRepositoryTests.java]: I opened https://github.com/elastic/elasticsearch/pull/46802

**Pull request merged**